### ISSUE TITLE
Published a plugin for pytest

### ISSUE THREAD

**Issue by studio3104** [aws/chalice#1120]
This is not an issue but just an announcement, I published a plugin just as titled. https://pypi.org/project/pytest-chalice/

I adopted the LocalGateway approach with reference to the methodology discussed in [this issue](https://github.com/aws/chalice/issues/289).
Also, exposed an abstract client fixture like `pytest-django` and `pytest-flask`, for example,

```python
from chalice import Chalice

app = Chalice(__name__)


@app.route('/')
def index:
    return {'hello': 'world'}
```

```python
from http import HTTPStatus


def test_index(client):
    response = client.get('/')
    assert response.status_code == HTTPStatus.OK
    assert response.json == {'hello': 'world'}
```

Any feedback is welcome, I'd be happy if you guys try it out, thanks.

**Comment by jamesls** [comment]
Nice!  What would you think about porting over some of that stuff to chalice directly?  I think the pytest fixtures would have to be separate, but things like the `InternalLocalGateway` and `ResponseHandler` could be added to chalice.

**Comment by studio3104** [comment]
Thanks for your suggestion, I'm more than happy to do that, going to make a PR to port them then!

**Comment by jamesls** [comment]
Closing, #1468 has been merged and is now available in the 1.17.0 release.  Thanks for your work on this!
